### PR TITLE
20240818 ec crypto

### LINKED ISSUE #142: BUG: Unable to load EC private key with rustls
When loading an EC private key the following error is raised:

```
Custom { kind: Other, error: "invalid certificate format, received: Sec1Key(PrivateSec1KeyDer(\"[secret key elided]\"))" }
```

This appears to be raised from https://docs.rs/axum-server/latest/src/axum_server/tls_rustls/mod.rs.html#345 

The match statement here may need to be extended, since https://docs.rs/rustls/latest/rustls/struct.ConfigBuilder.html#method.with_single_cert can accept a Sec1 Private key.

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #142 - allows usage of EC and PKCS1 keys.

### CODE REVIEW

**Review by daxpedda — CHANGES_REQUESTED** [review]
Could you add a changelog entry as well?

**Comment by Firstyear** [comment]
@daxpedda Done, I hope that helps :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ The format is based on [Keep a Changelog], and this project adheres to
 # Unreleased
 
 - **changed**: Updated `tower` from `0.4` to `0.5`.
+- **added**: Support `Pkcs1` and `Sec1` private keys with rustls.
```
Comment (daxpedda) [inline CHANGELOG.md:11]:
```suggestion
- **added**: Support reading PKCS\#1 and SEC1 private keys with Rustls.
```

**Comment by Firstyear** [comment]
Happy to help :)

**Pull request merged**